### PR TITLE
CTED-2309

### PR DESCRIPTION
**Pull request by BrendonPretorius**

### CODE REVIEW

**Review by Mmahadinana — APPROVED** [review]

File: classes/controller/forum_interaction_controller.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    public function viewstudentforuminteractiontrimes_action() {
+        global $PAGE, $OUTPUT;
+
+        $userview = 'student';
+        $roles = [
+            'student'
+        ];
+        $discussionid = optional_param('discussionid', null, PARAM_INT);
+
+        // TODO - find a better way for handling count sql returning no rows. For POC. just keeping it like this
```
Comment (JJSwanevelder) [inline classes/controller/forum_interaction_controller.php:126]: Please have BSA create ticket for this so we can track this issue

File: classes/controller/forum_interaction_controller.php [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        global $PAGE, $OUTPUT;
+
+        $userview = 'facilitator';
+        $roles = [
+            'tutor',
+            'headtutor'
+        ];
+        $discussionid = optional_param('discussionid', null, PARAM_INT);
+
+        // TODO - find a better way for handling count sql returning no rows. For POC. just keeping it like this
```
Comment (JJSwanevelder) [inline classes/controller/forum_interaction_controller.php:154]: Same as above

File: lib/table/time_interaction.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            $contextid
+        ];
+
+        if (!empty($discussionid)) {
+            $params[] = $discussionid;
+            $discussionsql = 'AND d.id = ?';
+        }
+        $params = array_merge($params, $roles);
+
+        // TODO - check cap types needed
```
Comment (JJSwanevelder) [inline lib/table/time_interaction.php:59]: Make sure ticket is created for this please

**Review by JJSwanevelder — APPROVED** [review]
👍

**Pull request merged**